app/graphql/types/query_type.rb

```diff
@@ -36,6 +41,15 @@ class QueryType < Types::BaseObject
       argument :category_id, ID, required: false
     end
 
+    def weather_map(id: nil)
+      return OpenWeatherMap.find(id) if id.present?
+
+      latest_weather_data = OpenWeatherMap.last
+      return latest_weather_data if latest_weather_data.present? && latest_weather_data.created_at > (Time.now - 1.hour)
```

Comment:
bitte kommentieren, dass hier irgendwie eine stunde gilt als cache oder so? und sonst neu importiert wird?

d.h. aber auch, dass wenn es keine daten der letzten stunde gibt, wird der importer angestoßen, sobald man eine query abschickt?
sorgt das nicht dafür, das es zu viele imports geben könnte? oder löst ein user einen import aus, das dauert etwas, und andere user haben dann die daten der stunde?
was passiert wenn zwei nutzer fast zeitgleich die anfrage machen?
hat das vorteile gegenüber einem cronjob, der stündlich laufen würde?

Reply:
Ein User löst einen Import aus, das dauert etwas,  andere user haben dann die Daten der stunde.
Wenn 2 User simultan den Import starten würde es unter umständen eine Anfrage zuviel geben. Durch die Abkapselung in eine Transaction ist der Stand in der DB aber abgesichert.

Vorteil:
- wir haben keine Spitzenlasten zu einer bestimmten Uhrzeit sondern es Verteilt sich über die Stunde
- wir brauchen keinen CronJobService um die Daten zu laden
- wir brauchen keinen separaten Microservice der die Daten zum Mainserver schiebt alle Stunde.

Reply:
und für die stunden, wo keine nutzer daten anfragt, brauchen wir keine daten von open weather abfragen!